### PR TITLE
release automation fixes

### PR DESCRIPTION
**Pull request by epgts**
Address some issues we discovered on first use for 1.12.0.

### CODE REVIEW

**Comment by rtwalker** [comment]
I think I missed adding 1.12.0 to the `upgradeable_from` list in #606, but we probably want the release automation to take care of that as well going forward.

https://github.com/timescale/timescaledb-toolkit/blob/7c690d7068465de000d9b610daa64771ad0f1a23/extension/timescaledb_toolkit.control#LL8

File: tools/release [hunk 1]
```diff
@@ -172,35 +185,101 @@ done
...
 $nop git show
 
-$nop git push
+# TODO Create a pull request instead?
+if $push; then
+    $nop git push origin main
+fi
+
+# File issue for release tasks that are not yet automated.
+gh issue create -R timescale/timescaledb-toolkit -F- -t "Release $VERSION" <<EOF
```
Comment (Smittyvb) [inline tools/release:258]: I think we would only want to do this if `$push` is set? And also it should have a `$nop` at the start?

Comment (epgts) [reply tools/release:258]: Right on both counts!  I'll address that later today.  Thanks!

**Comment by epgts** [comment]
> I think I missed adding 1.12.0 to the `upgradeable_from` list in #606, but we probably want the release automation to take care of that as well going forward.
> 
> https://github.com/timescale/timescaledb-toolkit/blob/7c690d7068465de000d9b610daa64771ad0f1a23/extension/timescaledb_toolkit.control#LL8

Good point!  We need it to be on main AND on release branch in case of cherry picks.  We very nearly released 1.12.1 that couldn't upgrade from 1.12.0!

So I'm thinking:
- add $VERSION to `upgradeable_from` in the final "prep for next release" phase
- assert that the previous version is listed in `upgradeable_from` in the first "create release" phase

Except...  what's the previous version!  When we're cutting 1.13.0, we need to assert that 1.12.N is in upgradeable_from...  but what N!  Do we enumerate the tags?  Pretty gross...

We'll have to think about this one some more.

```
if [ $patch -eq 0 ]; then
	# E.g. releasing 1.13.0 - how do we know all the patch releases of 1.12 to require in upgradeable_from ?
	# Pick the last from `git tag | grep $major.$(( minor - 1)) | version-sort`?
else
	# E.g. releasing 1.13.2 - this one's easy, we can require every entry from 0..N-1 exists:
	prev=$major.$minor.$(( patch - 1 ))
fi
```

**Comment by epgts** [comment]
I was thinking more about this earlier this morning and I realize now we only need to consider the cherry-pick case.  Full releases already have upgradeable_from fixed up.  It's only when cherry-picking that we might fail to cherry-pick the upgradeable_from bump.

I ran this by Smitty and he didn't say I was an idiot so maybe there's something to it.

Updating the example from above:

```
if [ $patch -eq 0 ]; then
	# E.g. releasing 1.13.0 - these are never cherrypicks and we automatically set upgradeable_from on main.
	# Pick the last from `git tag | grep $major.$(( minor - 1)) | version-sort`?
else
	# E.g. releasing 1.13.2 - this one might be a cherry-pick, so we better enforce it!
	prev=$major.$minor.$(( patch - 1 ))
fi
```

**Review by Smittyvb — APPROVED** [review]

**Review by thatzopoulos — APPROVED** [review]

**Review by WireBaron — APPROVED** [review]

**Pull request merged**